---
title: "Genes of interest"
output:
  html_document:
    df_print: paged
---


```{r}
# package
library(ape)
library(clusterProfiler)

#bring in phase1 allele frequencies
phase1.all.freqs<-read.csv(file = "~/Desktop/anoph.3.march.2020/phase1.all.freqs.csv")
#make column for id
phase1.all.freqs$id<-paste(phase1.all.freqs$chrom,phase1.all.freqs$POS)
#bring in dataframe with locality and environmental data for each individual phase1
phase1.alldata<-read.csv(file = "~/Desktop/anoph.3.march.2020/phase1.allvariables.csv")
#subset to only the variables we need (lat,long,hum,temp,precip)
meta<-phase1.alldata[,c(16,17,18,20,22)]
#subset only the unique lat/longs
metapops<-unique(meta)
#sort by latitude to match the order of the allele frequency files
sort.pops.phase1 <-metapops[order(metapops$latitude),]

## read in lfmm outlier table
lfmm.outlier.table <- read.csv("~/Downloads/vetted.lfmm.hum.env.csv")[,2:3]
#add separate chrom and pos columns
lfmm.outlier.table$chrom<-data.frame(do.call('rbind', strsplit(as.character(lfmm.outlier.table$id),' ',fixed=TRUE)))[,1]
lfmm.outlier.table$pos<-data.frame(do.call('rbind', strsplit(as.character(lfmm.outlier.table$id),' ',fixed=TRUE)))[,2]

## read in bayescan outlier table
bayescan.outlier.table <- read.csv("~/Downloads/vetted.bayescenv.hum.env.csv")[,2:3]
#add separate chrom and pos columns
bayescan.outlier.table$chrom<-data.frame(do.call('rbind', strsplit(as.character(bayescan.outlier.table$id),' ',fixed=TRUE)))[,1]
bayescan.outlier.table$pos<-data.frame(do.call('rbind', strsplit(as.character(bayescan.outlier.table$id),' ',fixed=TRUE)))[,2]

## read in genomic regions and attributes file
genomic_location <- read.gff("~/Downloads/Anopheles_gambiae.AgamP4.47.chr.gff3.gz")
```


```{r}
#keep only regions identified as "gene"
gen_regions <- genomic_location[genomic_location$type == "gene", ] # subset to only gene regions

#pull all genes queried
all.gene.id<-data.frame(do.call('rbind', strsplit(as.character(gen_regions$attributes),';',fixed=TRUE)))[,1]
all.genes<-data.frame(do.call('rbind', strsplit(as.character(all.gene.id),':',fixed=TRUE)))[,2]
#write.table(all.genes, "~/Downloads/all.genes.anopheles.gambiae.txt", row.names = F, col.names = F, quote = F)
```


```{r}
#define chroms
chroms <- c("2R", "2L", "3R", "3L","X") # chromosomes of interest

#initialize gene.list for lfmm
lfmm.gene.list<-data.frame()
#fill gene.list by matching location of genes of interest in gene feature regions by chromosomes
for (i in 1:length(chroms)){
  genloc <- gen_regions[gen_regions$seqid == chroms[i], ] # genome info per chromosome
  pos <- as.numeric(as.character(lfmm.outlier.table[lfmm.outlier.table$chrom == chroms[i],4])) # outlierFST position per chromosome
  for (w in 1:length(pos)) {
    if (nrow(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,]) !=0){
    lfmm.gene.list<-rbind(lfmm.gene.list, cbind(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,], pos[w]))
    }
  }
  cat("chromosome", i, "of", length(chroms), "finished\n\n")
}

#pull out gene ID as its own column
lfmm.gene.list$gene.id<-data.frame(do.call('rbind', strsplit(as.character(lfmm.gene.list$attributes),';',fixed=TRUE)))[,1]
lfmm.gene.list$gene<-data.frame(do.call('rbind', strsplit(as.character(lfmm.gene.list$gene.id),':',fixed=TRUE)))[,2]

#make SNP ID column
lfmm.gene.list$SNPid<-paste(lfmm.gene.list$seqid,lfmm.gene.list$`pos[w]`)

#subset
lfmm.gene.list<-lfmm.gene.list[,c("SNPid","gene")]

table(lfmm.gene.list$gene)[table(lfmm.gene.list$gene) >10]
hist(table(lfmm.gene.list$gene))
#write.table(unique(lfmm.gene.list$gene), "~/Downloads/lfmm.hum.gene.list.txt", row.names = F, col.names = F, quote = F)
#write.table(lfmm.gene.list, "~/Downloads/lfmm.hum.gene.snp.match.txt", row.names = F, col.names = T, quote = F)
```

```{r}
#initialize gene.list for bayescan
bayescan.gene.list<-data.frame()
#fill gene.list
for (i in 1:length(chroms)){
  genloc <- gen_regions[gen_regions$seqid == chroms[i], ] # genome info per chromosome
  pos <- as.numeric(as.character(bayescan.outlier.table[bayescan.outlier.table$chrom == chroms[i],4])) # outlierFST position per chromosome
  for (w in 1:length(pos)) {
    if (nrow(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,]) !=0){
      bayescan.gene.list<-rbind(bayescan.gene.list, cbind(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,], pos[w]))
    }
  }
  cat("chromosome", i, "of", length(chroms), "finished\n\n")
}

#pull out gene ID as its own column
bayescan.gene.list$gene.id<-data.frame(do.call('rbind', strsplit(as.character(bayescan.gene.list$attributes),';',fixed=TRUE)))[,1]
bayescan.gene.list$gene<-data.frame(do.call('rbind', strsplit(as.character(bayescan.gene.list$gene.id),':',fixed=TRUE)))[,2]

#make SNP ID column
bayescan.gene.list$SNPid<-paste(bayescan.gene.list$seqid,bayescan.gene.list$`pos[w]`)

#subset
bayescan.gene.list<-bayescan.gene.list[,c("SNPid","gene")]

table(bayescan.gene.list$gene)[table(bayescan.gene.list$gene) >10]
hist(table(bayescan.gene.list$gene))

#write.table(unique(bayescan.gene.list$gene), "~/Downloads/bayescan.hum.gene.list.txt", row.names = F, col.names = F, quote = F)
#write.table(bayescan.gene.list, "~/Downloads/bayescan.hum.gene.snp.match.txt", row.names = F, col.names = T, quote = F)
#look into how many genes overlap between the methods
intersect(unique(lfmm.gene.list$gene),unique(bayescan.gene.list$gene))
```

LFMM overrep testing
```{r}
#overrepresentation testing
#read in each GO term to gene map (col1=geneID,col2=GOterm)
mart.export<-read.table("~/Downloads/mart_export (3).txt", sep = "\t", header = T)

#calculate overrepresentation for lfmm hum
lfmm.hum.enriched<-enricher(gene = lfmm.gene.list$gene,
                 universe = all.genes,
                   pAdjustMethod = "fdr",
                 TERM2GENE= mart.export[,c(2,1)]
)
result<-lfmm.hum.enriched@result

lfmm.hum.enriched.sig<-result[result$p.adjust <= .05,]
lfmm.hum.enriched.sig
#write.table(lfmm.hum.enriched.sig[,c("ID","geneID")], file="~/Downloads/lfmm.hum.go.overrep.txt",
#            row.names = F, col.names = F, quote = F)

```

Bayescenv overrep testing
```{r}
#calculate overrepresentation for bayescan hum
bayescan.hum.enriched<-enricher(gene = bayescan.gene.list$gene,
                            universe = all.genes,
                            pAdjustMethod = "fdr",
                            TERM2GENE= mart.export[,c(2,1)]
                            )
result<-bayescan.hum.enriched@result

#retain significantly enriched GO terms
bayescan.hum.enriched.sig<-result[result$p.adjust <= .05,]

#check out significantly overrepresented GO terms
bayescan.hum.enriched.sig
```


```{r}
#get a vector of genes we found SNPs in for each enriched GO term
#lfmm
go.0055085<-unlist(strsplit(as.character(lfmm.hum.enriched.sig$geneID[lfmm.hum.enriched.sig$ID == "GO:0055085"]),'/'))
go.0004672<-unlist(strsplit(as.character(lfmm.hum.enriched.sig$geneID[lfmm.hum.enriched.sig$ID == "GO:0004672"]),'/'))
go.0010951<-unlist(strsplit(as.character(lfmm.hum.enriched.sig$geneID[lfmm.hum.enriched.sig$ID == "GO:0010951"]),'/'))
go.0005509<-unlist(strsplit(as.character(lfmm.hum.enriched.sig$geneID[lfmm.hum.enriched.sig$ID == "GO:0005509"]),'/'))
go.0006468<-unlist(strsplit(as.character(lfmm.hum.enriched.sig$geneID[lfmm.hum.enriched.sig$ID == "GO:0006468"]),'/'))
#bayescan
go.0004930<-unlist(strsplit(as.character(bayescan.hum.enriched.sig$geneID[bayescan.hum.enriched.sig$ID == "GO:0004930"]),'/'))
go.0007186<-unlist(strsplit(as.character(bayescan.hum.enriched.sig$geneID[bayescan.hum.enriched.sig$ID == "GO:0007186"]),'/'))
go.0006368<-unlist(strsplit(as.character(bayescan.hum.enriched.sig$geneID[bayescan.hum.enriched.sig$ID == "GO:0006368"]),'/'))
```

```{r}
#take them 1 by 1
#GO:0055085 = transmembrane transport
#The process in which a solute is transported across a lipid bilayer, from one side of a membrane to the other.
go.0055085

#make subset list of SNPs and genes associated with this specific GO term
lfmm.gene.list.go.0055085<-lfmm.gene.list[lfmm.gene.list$gene %in% go.0055085,]
#This is a set of genes and SNPs found all over the genome
table(lfmm.gene.list[lfmm.gene.list$gene %in% go.0055085,]$gene)[table(lfmm.gene.list[lfmm.gene.list$gene %in% go.0055085,]$gene) >0]
#a few genes have many SNPs and some have only 1
```

plot all SNPs associated w/ GO:055085
```{r}
#plot all SNPs associated w/ GO:0055085
sig.pos<-lfmm.gene.list.go.0055085$SNPid
gene<-droplevels(lfmm.gene.list.go.0055085$gene)
#plot allele freqs vs humidity for these SNPs
par(mfrow=c(3,3))
for (i in 1:length(sig.pos)){
  frq<-as.numeric(as.vector(phase1.all.freqs[phase1.all.freqs$id == sig.pos[i],3:16]))
  plot(sort.pops.phase1$hannual,frq, main = gene[i], ylim = c(0,1), xlab=sig.pos[i])
  abline(lm(frq~sort.pops.phase1$hannual))
}
```

#Best places to find info:
#uniprot database:
#uniprot.org/
#UC Irvine AGAM Gene Expression Profile
#http://www.angagepuci.bio.uci.edu/
#AG1000G selection atlas
#https://malariagen.github.io/agam-selection-atlas/0.1-alpha3/index.html

look into each gene
AGAP001966 (2R 12980530) "urea transporter activity" expressed strongly in males
http://www.angagepuci.bio.uci.edu/lookup2.php?geneid=Ag.2R.2140.0_CDS_a_at

AGAP002331 (2R 19959027) 2 SNPs similar pattern "sulfate transporter"
http://www.angaged.bio.uci.edu/lookup2.php?geneid=Ag.2R.2175.0_CDS_at

AGAP002369 (2R 20722346) "acetylcholine transmembrane transporter activity"
https://www.uniprot.org/uniprot/Q7PR83

AGAP002578 (2R 2316854) 5 SNPs with similar pattern. Associated with courtship in Drosophila
http://www.angagepuci.bio.uci.edu/lookup2.php?geneid=Ag.2R.3261.0_CDS_at

AGAP002826 (2R 28045726) 99.9% selection signal in Uganda (gambiae) and 100% selection signal Angola (coluzzii)
strong selective sweep in Uganda (gambiae) on 2R between positions 27,980,000 and 28,280,000
23 genes in this sweep including some candidate insecticide resistance genes
https://malariagen.github.io/agam-selection-atlas/0.1-alpha3/signal/XPEHH/UGS.BFS/2/1/index.html
in Angola (coluzzii) strong and narrow selective sweep from 2R 28,040,000 and 28,080,000
4 genes in focal region including one candidate insecticide resistance gene
https://malariagen.github.io/agam-selection-atlas/0.1-alpha3/signal/H12/AOM/2/2/index.html

AGAP002859 (2R 28405107) one of 11 genes within the Cyp6p locus (known insecticide resistance)
(solute carrier family 8 (sodium/calcium exchanger)) We recover 4 SNPs with unique humidity associated patterns
Müller et al. (2008) Field-Caught Permethrin-Resistant Anopheles gambiae Overexpress CYP6P3, a P450 That Metabolises Pyrethroids
cyp6p is 2R: 28480576 to 28505816
```{r}
lfmm.outlier.table[lfmm.outlier.table$chrom == "2R" & as.numeric(lfmm.outlier.table$pos) > 28480576  & as.numeric(lfmm.outlier.table$pos) < 28505816,]
```

we actually don't have any outlier env correlated SNPs in cyp6p, we have 4 environmentally correlated SNPs upstream of the
strictly defined locus in AGAP002859 gene which overlaps into cyp6p locus.
this potentially means that our env signal is not being confounded by massive selection pressure for insecticide resistance.
AGAP002859 is a huge gene with the entire Cyp6p locus within it. Need to research more to figure out exactly what is going on here

AGAP001634 (2R 7199475) SODIUM SOLUTE SYMPORTER AEDES AEGYPTI
http://www.angagepuci.bio.uci.edu/lookup2.php?geneid=Ag.2R.3089.0_CDS_at

AGAP006347 (2R 30242271) potassium voltage-gated channel KQT-like subfamily, invertebrate [Source:VB Community Annotation]
3 unique SNPs w/ env correlations. large gene many introns
https://www.vectorbase.org/Anopheles_gambiae/Gene/Summary?g=AGAP006347;r=2L:30175822-30340954

AGAP007086 (2L 42306780): Sodium channel protein, identified as a "target-site" insecticide resistance candidate gene
95.5% selection in Burkina Faso (gambiae). We have 11 SNPs w/ concordant env associations

AGAP009123 (3R 2640917): MONOCARBOXYLATE TRANSPORTER AEDES AEGYPTI 
http://www.angagepuci.bio.uci.edu/lookup2.php?geneid=Ag.3R.2395.0_CDS_at

AGAP009188 (3R 28574369): (solute carrier family 39 (zinc transporter), member 9)
This is directly upstream of the Gste locus (insecticide resistance conferring) but this specific gene is not implicated
Mitchell et al. (2014) Metabolic and Target-Site Mechanisms Combine to Confer Strong DDT Resistance in Anopheles gambiae

AGAP007732 (3R 36361): 11 linked SNPs. pattern kind of a mess.
	SODIUM-DEPENDENT PHOSPHATE TRANSPORTER AEDES AEGYPTI SODIUM DEPENDENT DROSOPHILA PSEUDOOBSCURA 
http://www.angagepuci.bio.uci.edu/lookup2.php?geneid=Ag.3R.652.0_UTR_at

AGAP007733 (3R 39675): 3 more SNPs with concordant pattern.
No selection information. 	SODIUM-DEPENDENT PHOSPHATE TRANSPORTER AEDES AEGYPTI 
http://www.angagepuci.bio.uci.edu/lookup2.php?geneid=Ag.3R.1156.0_CDS_at

AGAP009835 (3R 44069554): Vacuolar glutathione S-conjugate transporter of the ATP-binding cassette family, role in detoxifying metals
>99% distribution selection response in Angola (coluzzii) and Cameroon (gambiae)
https://malariagen.github.io/agam-selection-atlas/0.1-alpha3/gene/AGAP009835.html?highlight=agap009835

AGAP010269 (3R 5188731): ion channel activity
https://www.uniprot.org/uniprot/Q7Q0M1

AGAP029240 (3R 6713749): little info available

AGAP010794 (3L 10429259): cation transmembrane transporter activity (cell membrane)
https://www.uniprot.org/uniprot/Q7Q231
weakly selected in Gabon (gambiae)
https://malariagen.github.io/agam-selection-atlas/0.1-alpha3/signal/IHS/GAS/3/2/index.html

AGAP010854 (3L 12057162): sucrose proton symporter
https://www.uniprot.org/uniprot/Q7QGQ4

AGAP029297 (3L 12061750): little info online

AGAP010383 (3L 2417833): cell wall membrane protein
https://www.uniprot.org/uniprot/Q7PRI8

AGAP029048 (3L 255858886): plasma membrane protein (cellular hypotonic salinity response)
https://www.uniprot.org/uniprot/A0A1Y9IHS4

AGAP012154 (3L 38088469): transmembrane peptide transport
https://www.uniprot.org/uniprot/Q7PZY8

AGAP010562 (3L 6394642): solute carrier family 7 (cationic amino acid transporter, y+ system)
http://www.angagepuci.bio.uci.edu/lookup2.php?geneid=Ag.3L.698.1_CDS_at

AGAP000795 (X 14516591):Protein with similarity to mammalian monocarboxylate permeases
which are involved in transport of monocarboxylic acids across the plasma membrane 
98.5% percentile for selection in Gabon (gambiae) and Angola (coluzzii)
https://malariagen.github.io/agam-selection-atlas/0.1-alpha3/gene/AGAP000795.html?highlight=agap000795


#lfmm
```{r}
go.0055085
go.0004672
go.0010951
go.0005509
go.0006468
```

#bayescan
```{r}
go.0004930
go.0007186
go.0006368
```


