---
title: "Gene.overlap.investigation"
author: "Devon DeRaad"
date: "6/30/2020"
output: 
  html_document:
    self_contained: false
---

```{r}
#May 17 2020
#plot results from phase1 and verify with phase2 allele freq patterns
devtools::install_github('exaexa/scattermore')
library(scattermore)
library(gridExtra)
library(VennDiagram)
library(qvalue)
library(ggplot2)

#bring in all sigs
all.sigs<-read.csv(file="~/Desktop/anoph.3.march.2020/all.sigs.csv")

#bring in gene list by variable and method
#hum
lfmm.hum<-read.csv("~/Downloads/lfmm.hum.gene.list.txt", header=F)
bayescenv.hum<-read.csv("~/Downloads/bayescan.hum.gene.list.txt", header=F)
#prec
lfmm.prec<-read.csv("~/Downloads/lfmm.prec.gene.list.txt", header=F)
bayescenv.prec<-read.csv("~/Downloads/bayescan.prec.gene.list.txt", header=F)
#temp
lfmm.temp<-read.csv("~/Downloads/lfmm.temp.gene.list.txt", header=F)
bayescenv.temp<-read.csv("~/Downloads/bayescan.temp.gene.list.txt", header=F)

```


```{r}
#compare lfmm and bayescenv temps
plot1<-ggplot(data=all.sigs, aes(x=-log10(temp.q), y=-log10(temp.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("bayescan Q")+ylab("LFMM P")+ggtitle("temperature")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#compare lfmm and bayescenv prec
plot2<-ggplot(data=all.sigs, aes(x=-log10(prec.q), y=-log10(prec.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("bayescan Q")+ylab("LFMM P")+ggtitle("prec")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#compare lfmm and bayescenv hum
plot3<-ggplot(data=all.sigs, aes(x=-log10(hum.q), y=-log10(hum.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("bayescan Q")+ylab("LFMM P")+ggtitle("hum")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#plot them all together
grid.arrange(plot1,plot2,plot3, nrow=1)
```

```{r}
#compare lfmm and bayescenv temps
plot1<-ggplot(data=all.sigs, aes(x=-log10(hum.p), y=-log10(prec.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("hum")+ylab("prec")+ggtitle("lfmm variable comparison")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#compare lfmm and bayescenv prec
plot2<-ggplot(data=all.sigs, aes(x=-log10(hum.p), y=-log10(temp.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("hum")+ylab("temp")+ggtitle("lfmm variable comp")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#compare lfmm and bayescenv hum
plot3<-ggplot(data=all.sigs, aes(x=-log10(prec.p), y=-log10(temp.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("prec")+ylab("temp")+ggtitle("lfmm variable comp")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#plot them all together
grid.arrange(plot1,plot2,plot3, nrow=1)
```

```{r}
#compare lfmm and bayescenv temps
plot1<-ggplot(data=all.sigs, aes(x=-log10(hum.q), y=-log10(prec.q)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("hum")+ylab("prec")+ggtitle("baye variable comparison")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#compare lfmm and bayescenv prec
plot2<-ggplot(data=all.sigs, aes(x=-log10(hum.q), y=-log10(temp.q)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("hum")+ylab("temp")+ggtitle("baye variable comp")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#compare lfmm and bayescenv hum
plot3<-ggplot(data=all.sigs, aes(x=-log10(prec.q), y=-log10(temp.q)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab("prec")+ylab("temp")+ggtitle("baye variable comp")+
  geom_vline(xintercept=c(-log10(.01)), linetype=c("dashed"))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c("dashed"))
#plot them all together
grid.arrange(plot1,plot2,plot3, nrow=1)
```


```{r}
venn.diagram(x = list(lfmm.hum$V1,
                      lfmm.prec$V1,
                      lfmm.temp$V1),
             category.names = c("lfmm hum", "lfmm prec", "lfmm temp"),
             main = "genes containing lfmm vetted outlier SNPs",
             filename = "~/Downloads/lfmm.allvars.png")
```

```{r, echo=FALSE,out.width="49%", out.height="20%",fig.show='hold',fig.align='center'}
knitr::include_graphics("~/Downloads/lfmm.allvars.png")
```


```{r}
venn.diagram(x = list(bayescenv.hum$V1,
                      bayescenv.prec$V1,
                      bayescenv.temp$V1),
             category.names = c("bayescenv hum", "bayescenv prec", "bayescenv temp"),
             main = "genes containing bayescenv vetted outlier SNPs",
             filename = "~/Downloads/bayescenv.allvars.png")
```

```{r, echo=FALSE,out.width="49%", out.height="20%",fig.show='hold',fig.align='center'}
knitr::include_graphics("~/Downloads/bayescenv.allvars.png")
```


```{r}
#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(bayescenv.hum$V1,lfmm.hum$V1),
             category.names = c("baye hum", "lfmm hum"),
             main = "genes containing vetted outlier hum SNPs",
             filename = "~/Downloads/hum.overlap.png", euler.d=FALSE,scaled=FALSE)

```

```{r, echo=FALSE,out.width="49%", out.height="20%",fig.show='hold',fig.align='center'}
knitr::include_graphics("~/Downloads/hum.overlap.png")
```

```{r}
#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(bayescenv.prec$V1,lfmm.prec$V1),
             category.names = c("baye prec", "lfmm prec"),
             main = "genes containing vetted outlier prec SNPs",
             filename = "~/Downloads/prec.overlap.png", euler.d=FALSE,scaled=FALSE)

```

```{r, echo=FALSE,out.width="49%", out.height="20%",fig.show='hold',fig.align='center'}
knitr::include_graphics("~/Downloads/prec.overlap.png")
```


```{r}
#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(bayescenv.temp$V1,lfmm.temp$V1),
             category.names = c("baye temp", "lfmm temp"),
             main = "genes containing vetted outlier temp SNPs",
             filename = "~/Downloads/temp.overlap.png", euler.d=FALSE,scaled=FALSE)

```

```{r, echo=FALSE,out.width="49%", out.height="20%",fig.show='hold',fig.align='center'}
knitr::include_graphics("~/Downloads/temp.overlap.png")
```

```{r}
#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(intersect(unique(lfmm.hum$V1),unique(bayescenv.hum$V1)),
                      intersect(unique(lfmm.prec$V1),unique(bayescenv.prec$V1)),
                      intersect(unique(lfmm.temp$V1),unique(bayescenv.temp$V1))),
             category.names = c("hum", "prec", "temp"),
             main = "genes w/ outlier SNPs in both methods",
             filename = "~/Downloads/snps.png", euler.d=FALSE,scaled=FALSE)
```

```{r, echo=FALSE,out.width="49%", out.height="20%",fig.show='hold',fig.align='center'}
knitr::include_graphics("~/Downloads/snps.png")
```

```{r}
intersect(unique(lfmm.hum$V1),unique(bayescenv.hum$V1))
intersect(unique(lfmm.prec$V1),unique(bayescenv.prec$V1))
intersect(unique(lfmm.temp$V1),unique(bayescenv.temp$V1))
```



