<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Devon DeRaad" />


<title>Gene.overlap.investigation</title>

<script src="gene.overlap.investigation_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="gene.overlap.investigation_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="gene.overlap.investigation_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="gene.overlap.investigation_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="gene.overlap.investigation_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="gene.overlap.investigation_files/navigation-1.1/tabsets.js"></script>
<link href="gene.overlap.investigation_files/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="gene.overlap.investigation_files/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene.overlap.investigation</h1>
<h4 class="author">Devon DeRaad</h4>
<h4 class="date">6/30/2020</h4>

</div>


<pre class="r"><code>#May 17 2020
#plot results from phase1 and verify with phase2 allele freq patterns
devtools::install_github(&#39;exaexa/scattermore&#39;)</code></pre>
<pre><code>## Skipping install of &#39;scattermore&#39; from a github remote, the SHA1 (a69dc9b9) has not changed since last install.
##   Use `force = TRUE` to force installation</code></pre>
<pre class="r"><code>library(scattermore)
library(gridExtra)
library(VennDiagram)</code></pre>
<pre><code>## Loading required package: grid</code></pre>
<pre><code>## Loading required package: futile.logger</code></pre>
<pre class="r"><code>library(qvalue)</code></pre>
<pre><code>## Warning: package &#39;qvalue&#39; was built under R version 3.5.2</code></pre>
<pre class="r"><code>library(ggplot2)

#bring in all sigs
all.sigs&lt;-read.csv(file=&quot;~/Desktop/anoph.3.march.2020/all.sigs.csv&quot;)

#bring in gene list by variable and method
#hum
lfmm.hum&lt;-read.csv(&quot;~/Downloads/lfmm.hum.gene.list.txt&quot;, header=F)
bayescenv.hum&lt;-read.csv(&quot;~/Downloads/bayescan.hum.gene.list.txt&quot;, header=F)
#prec
lfmm.prec&lt;-read.csv(&quot;~/Downloads/lfmm.prec.gene.list.txt&quot;, header=F)
bayescenv.prec&lt;-read.csv(&quot;~/Downloads/bayescan.prec.gene.list.txt&quot;, header=F)
#temp
lfmm.temp&lt;-read.csv(&quot;~/Downloads/lfmm.temp.gene.list.txt&quot;, header=F)
bayescenv.temp&lt;-read.csv(&quot;~/Downloads/bayescan.temp.gene.list.txt&quot;, header=F)</code></pre>
<pre class="r"><code>#compare lfmm and bayescenv temps
plot1&lt;-ggplot(data=all.sigs, aes(x=-log10(temp.q), y=-log10(temp.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;bayescan Q&quot;)+ylab(&quot;LFMM P&quot;)+ggtitle(&quot;temperature&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#compare lfmm and bayescenv prec
plot2&lt;-ggplot(data=all.sigs, aes(x=-log10(prec.q), y=-log10(prec.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;bayescan Q&quot;)+ylab(&quot;LFMM P&quot;)+ggtitle(&quot;prec&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#compare lfmm and bayescenv hum
plot3&lt;-ggplot(data=all.sigs, aes(x=-log10(hum.q), y=-log10(hum.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;bayescan Q&quot;)+ylab(&quot;LFMM P&quot;)+ggtitle(&quot;hum&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#plot them all together
grid.arrange(plot1,plot2,plot3, nrow=1)</code></pre>
<p><img src="gene.overlap.investigation_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code>#compare lfmm and bayescenv temps
plot1&lt;-ggplot(data=all.sigs, aes(x=-log10(hum.p), y=-log10(prec.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;hum&quot;)+ylab(&quot;prec&quot;)+ggtitle(&quot;lfmm variable comparison&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#compare lfmm and bayescenv prec
plot2&lt;-ggplot(data=all.sigs, aes(x=-log10(hum.p), y=-log10(temp.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;hum&quot;)+ylab(&quot;temp&quot;)+ggtitle(&quot;lfmm variable comp&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#compare lfmm and bayescenv hum
plot3&lt;-ggplot(data=all.sigs, aes(x=-log10(prec.p), y=-log10(temp.p)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;prec&quot;)+ylab(&quot;temp&quot;)+ggtitle(&quot;lfmm variable comp&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#plot them all together
grid.arrange(plot1,plot2,plot3, nrow=1)</code></pre>
<p><img src="gene.overlap.investigation_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<pre class="r"><code>#compare lfmm and bayescenv temps
plot1&lt;-ggplot(data=all.sigs, aes(x=-log10(hum.q), y=-log10(prec.q)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;hum&quot;)+ylab(&quot;prec&quot;)+ggtitle(&quot;baye variable comparison&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#compare lfmm and bayescenv prec
plot2&lt;-ggplot(data=all.sigs, aes(x=-log10(hum.q), y=-log10(temp.q)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;hum&quot;)+ylab(&quot;temp&quot;)+ggtitle(&quot;baye variable comp&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#compare lfmm and bayescenv hum
plot3&lt;-ggplot(data=all.sigs, aes(x=-log10(prec.q), y=-log10(temp.q)))+
  geom_scattermore(pointsize = 1.2) +theme_classic()+xlab(&quot;prec&quot;)+ylab(&quot;temp&quot;)+ggtitle(&quot;baye variable comp&quot;)+
  geom_vline(xintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))+
  geom_hline(yintercept=c(-log10(.01)), linetype=c(&quot;dashed&quot;))
#plot them all together
grid.arrange(plot1,plot2,plot3, nrow=1)</code></pre>
<p><img src="gene.overlap.investigation_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>venn.diagram(x = list(lfmm.hum$V1,
                      lfmm.prec$V1,
                      lfmm.temp$V1),
             category.names = c(&quot;lfmm hum&quot;, &quot;lfmm prec&quot;, &quot;lfmm temp&quot;),
             main = &quot;genes containing lfmm vetted outlier SNPs&quot;,
             filename = &quot;~/Downloads/lfmm.allvars.png&quot;)</code></pre>
<pre><code>## [1] 1</code></pre>
<p><img src="lfmm.allvars.png" width="49%" height="20%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>venn.diagram(x = list(bayescenv.hum$V1,
                      bayescenv.prec$V1,
                      bayescenv.temp$V1),
             category.names = c(&quot;bayescenv hum&quot;, &quot;bayescenv prec&quot;, &quot;bayescenv temp&quot;),
             main = &quot;genes containing bayescenv vetted outlier SNPs&quot;,
             filename = &quot;~/Downloads/bayescenv.allvars.png&quot;)</code></pre>
<pre><code>## [1] 1</code></pre>
<p><img src="bayescenv.allvars.png" width="49%" height="20%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(bayescenv.hum$V1,lfmm.hum$V1),
             category.names = c(&quot;baye hum&quot;, &quot;lfmm hum&quot;),
             main = &quot;genes containing vetted outlier hum SNPs&quot;,
             filename = &quot;~/Downloads/hum.overlap.png&quot;, euler.d=FALSE,scaled=FALSE)</code></pre>
<pre><code>## [1] 1</code></pre>
<p><img src="hum.overlap.png" width="49%" height="20%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(bayescenv.prec$V1,lfmm.prec$V1),
             category.names = c(&quot;baye prec&quot;, &quot;lfmm prec&quot;),
             main = &quot;genes containing vetted outlier prec SNPs&quot;,
             filename = &quot;~/Downloads/prec.overlap.png&quot;, euler.d=FALSE,scaled=FALSE)</code></pre>
<pre><code>## [1] 1</code></pre>
<p><img src="prec.overlap.png" width="49%" height="20%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(bayescenv.temp$V1,lfmm.temp$V1),
             category.names = c(&quot;baye temp&quot;, &quot;lfmm temp&quot;),
             main = &quot;genes containing vetted outlier temp SNPs&quot;,
             filename = &quot;~/Downloads/temp.overlap.png&quot;, euler.d=FALSE,scaled=FALSE)</code></pre>
<pre><code>## [1] 1</code></pre>
<p><img src="temp.overlap.png" width="49%" height="20%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot venn diagrams to compare the snps identified by lfmm for the three variables
venn.diagram(x = list(intersect(unique(lfmm.hum$V1),unique(bayescenv.hum$V1)),
                      intersect(unique(lfmm.prec$V1),unique(bayescenv.prec$V1)),
                      intersect(unique(lfmm.temp$V1),unique(bayescenv.temp$V1))),
             category.names = c(&quot;hum&quot;, &quot;prec&quot;, &quot;temp&quot;),
             main = &quot;genes w/ outlier SNPs in both methods&quot;,
             filename = &quot;~/Downloads/snps.png&quot;, euler.d=FALSE,scaled=FALSE)</code></pre>
<pre><code>## [1] 1</code></pre>
<p><img src="snps.png" width="49%" height="20%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>intersect(unique(lfmm.hum$V1),unique(bayescenv.hum$V1))</code></pre>
<pre><code>##  [1] &quot;AGAP002310&quot; &quot;AGAP002336&quot; &quot;AGAP002369&quot; &quot;AGAP002573&quot; &quot;AGAP002578&quot;
##  [6] &quot;AGAP002649&quot; &quot;AGAP002678&quot; &quot;AGAP002741&quot; &quot;AGAP002748&quot; &quot;AGAP002822&quot;
## [11] &quot;AGAP002826&quot; &quot;AGAP002858&quot; &quot;AGAP002859&quot; &quot;AGAP002879&quot; &quot;AGAP002883&quot;
## [16] &quot;AGAP002896&quot; &quot;AGAP003632&quot; &quot;AGAP003633&quot; &quot;AGAP004767&quot; &quot;AGAP009201&quot;
## [21] &quot;AGAP009210&quot; &quot;AGAP010817&quot; &quot;AGAP010820&quot; &quot;AGAP029564&quot; &quot;AGAP029616&quot;
## [26] &quot;AGAP000821&quot; &quot;AGAP000830&quot; &quot;AGAP000519&quot;</code></pre>
<pre class="r"><code>intersect(unique(lfmm.prec$V1),unique(bayescenv.prec$V1))</code></pre>
<pre><code>##  [1] &quot;AGAP001824&quot; &quot;AGAP029626&quot; &quot;AGAP002916&quot; &quot;AGAP029619&quot; &quot;AGAP001674&quot;
##  [6] &quot;AGAP004707&quot; &quot;AGAP006347&quot; &quot;AGAP006590&quot; &quot;AGAP029578&quot; &quot;AGAP007736&quot;
## [11] &quot;AGAP007754&quot; &quot;AGAP007761&quot; &quot;AGAP007762&quot; &quot;AGAP007763&quot; &quot;AGAP010816&quot;
## [16] &quot;AGAP010817&quot; &quot;AGAP000844&quot; &quot;AGAP000847&quot; &quot;AGAP029672&quot; &quot;AGAP000962&quot;
## [21] &quot;AGAP000974&quot; &quot;AGAP001022&quot; &quot;AGAP001023&quot; &quot;AGAP001025&quot; &quot;AGAP001029&quot;
## [26] &quot;AGAP001031&quot; &quot;AGAP001034&quot; &quot;AGAP001035&quot; &quot;AGAP001036&quot; &quot;AGAP001037&quot;
## [31] &quot;AGAP001038&quot; &quot;AGAP001039&quot; &quot;AGAP001040&quot; &quot;AGAP001043&quot; &quot;AGAP001046&quot;
## [36] &quot;AGAP001048&quot; &quot;AGAP001052&quot; &quot;AGAP001053&quot; &quot;AGAP013021&quot; &quot;AGAP001061&quot;
## [41] &quot;AGAP001064&quot; &quot;AGAP001065&quot; &quot;AGAP001069&quot; &quot;AGAP001070&quot; &quot;AGAP001073&quot;
## [46] &quot;AGAP001076&quot; &quot;AGAP001082&quot; &quot;AGAP001083&quot; &quot;AGAP001084&quot; &quot;AGAP001091&quot;
## [51] &quot;AGAP001092&quot; &quot;AGAP001093&quot; &quot;AGAP001094&quot;</code></pre>
<pre class="r"><code>intersect(unique(lfmm.temp$V1),unique(bayescenv.temp$V1))</code></pre>
<pre><code>##   [1] &quot;AGAP002444&quot; &quot;AGAP002449&quot; &quot;AGAP002817&quot; &quot;AGAP002824&quot; &quot;AGAP002832&quot;
##   [6] &quot;AGAP002835&quot; &quot;AGAP002845&quot; &quot;AGAP002846&quot; &quot;AGAP002857&quot; &quot;AGAP002858&quot;
##  [11] &quot;AGAP002859&quot; &quot;AGAP002869&quot; &quot;AGAP000586&quot; &quot;AGAP029190&quot; &quot;AGAP002874&quot;
##  [16] &quot;AGAP002875&quot; &quot;AGAP002876&quot; &quot;AGAP002879&quot; &quot;AGAP002880&quot; &quot;AGAP002881&quot;
##  [21] &quot;AGAP002883&quot; &quot;AGAP002886&quot; &quot;AGAP002888&quot; &quot;AGAP002892&quot; &quot;AGAP003623&quot;
##  [26] &quot;AGAP003632&quot; &quot;AGAP003633&quot; &quot;AGAP003643&quot; &quot;AGAP003651&quot; &quot;AGAP003654&quot;
##  [31] &quot;AGAP003656&quot; &quot;AGAP003658&quot; &quot;AGAP004644&quot; &quot;AGAP013143&quot; &quot;AGAP004671&quot;
##  [36] &quot;AGAP004672&quot; &quot;AGAP004689&quot; &quot;AGAP004691&quot; &quot;AGAP004692&quot; &quot;AGAP004693&quot;
##  [41] &quot;AGAP004694&quot; &quot;AGAP004677&quot; &quot;AGAP028434&quot; &quot;AGAP029368&quot; &quot;AGAP004696&quot;
##  [46] &quot;AGAP029667&quot; &quot;AGAP028436&quot; &quot;AGAP028437&quot; &quot;AGAP004707&quot; &quot;AGAP004714&quot;
##  [51] &quot;AGAP004718&quot; &quot;AGAP004723&quot; &quot;AGAP004725&quot; &quot;AGAP004726&quot; &quot;AGAP004727&quot;
##  [56] &quot;AGAP004728&quot; &quot;AGAP004729&quot; &quot;AGAP029478&quot; &quot;AGAP004731&quot; &quot;AGAP029113&quot;
##  [61] &quot;AGAP004742&quot; &quot;AGAP004746&quot; &quot;AGAP004756&quot; &quot;AGAP004758&quot; &quot;AGAP004761&quot;
##  [66] &quot;AGAP004766&quot; &quot;AGAP004767&quot; &quot;AGAP004768&quot; &quot;AGAP004769&quot; &quot;AGAP004772&quot;
##  [71] &quot;AGAP004775&quot; &quot;AGAP004781&quot; &quot;AGAP004783&quot; &quot;AGAP004792&quot; &quot;AGAP004795&quot;
##  [76] &quot;AGAP004797&quot; &quot;AGAP004800&quot; &quot;AGAP004801&quot; &quot;AGAP004805&quot; &quot;AGAP004808&quot;
##  [81] &quot;AGAP004817&quot; &quot;AGAP004819&quot; &quot;AGAP004823&quot; &quot;AGAP004824&quot; &quot;AGAP004825&quot;
##  [86] &quot;AGAP004834&quot; &quot;AGAP004682&quot; &quot;AGAP004845&quot; &quot;AGAP004847&quot; &quot;AGAP004861&quot;
##  [91] &quot;AGAP004866&quot; &quot;AGAP029807&quot; &quot;AGAP004871&quot; &quot;AGAP004873&quot; &quot;AGAP004877&quot;
##  [96] &quot;AGAP004881&quot; &quot;AGAP004890&quot; &quot;AGAP004892&quot; &quot;AGAP028432&quot; &quot;AGAP009183&quot;
## [101] &quot;AGAP009189&quot; &quot;AGAP009191&quot; &quot;AGAP009195&quot; &quot;AGAP009196&quot; &quot;AGAP009198&quot;
## [106] &quot;AGAP009200&quot; &quot;AGAP009201&quot; &quot;AGAP009202&quot; &quot;AGAP029480&quot; &quot;AGAP009206&quot;
## [111] &quot;AGAP029511&quot; &quot;AGAP009210&quot; &quot;AGAP009212&quot; &quot;AGAP009213&quot; &quot;AGAP009224&quot;
## [116] &quot;AGAP009245&quot; &quot;AGAP009389&quot; &quot;AGAP029641&quot; &quot;AGAP009708&quot; &quot;AGAP009723&quot;
## [121] &quot;AGAP029589&quot; &quot;AGAP009744&quot; &quot;AGAP009757&quot; &quot;AGAP009770&quot; &quot;AGAP009784&quot;
## [126] &quot;AGAP009789&quot; &quot;AGAP009793&quot; &quot;AGAP029125&quot; &quot;AGAP029567&quot; &quot;AGAP009799&quot;
## [131] &quot;AGAP029238&quot; &quot;AGAP009805&quot; &quot;AGAP009809&quot; &quot;AGAP029648&quot; &quot;AGAP009860&quot;
## [136] &quot;AGAP009896&quot; &quot;AGAP009925&quot; &quot;AGAP009947&quot; &quot;AGAP010003&quot; &quot;AGAP010026&quot;
## [141] &quot;AGAP010090&quot; &quot;AGAP010147&quot; &quot;AGAP010158&quot; &quot;AGAP010160&quot; &quot;AGAP010162&quot;
## [146] &quot;AGAP010184&quot; &quot;AGAP010186&quot; &quot;AGAP010229&quot; &quot;AGAP010265&quot; &quot;AGAP010283&quot;
## [151] &quot;AGAP010286&quot; &quot;AGAP010290&quot; &quot;AGAP010293&quot; &quot;AGAP010294&quot; &quot;AGAP010295&quot;
## [156] &quot;AGAP010297&quot; &quot;AGAP010302&quot; &quot;AGAP010303&quot; &quot;AGAP010304&quot; &quot;AGAP029542&quot;
## [161] &quot;AGAP008173&quot; &quot;AGAP007736&quot; &quot;AGAP010330&quot; &quot;AGAP029651&quot; &quot;AGAP010782&quot;
## [166] &quot;AGAP010793&quot; &quot;AGAP010794&quot; &quot;AGAP029620&quot; &quot;AGAP029564&quot; &quot;AGAP010311&quot;
## [171] &quot;AGAP010990&quot; &quot;AGAP010312&quot; &quot;AGAP010394&quot; &quot;AGAP010410&quot; &quot;AGAP010313&quot;
## [176] &quot;AGAP029616&quot; &quot;AGAP010509&quot; &quot;AGAP010310&quot; &quot;AGAP010598&quot; &quot;AGAP010599&quot;
## [181] &quot;AGAP029596&quot; &quot;AGAP010691&quot; &quot;AGAP010740&quot; &quot;AGAP029636&quot; &quot;AGAP029471&quot;
## [186] &quot;AGAP010750&quot; &quot;AGAP013356&quot; &quot;AGAP000801&quot; &quot;AGAP000807&quot; &quot;AGAP000812&quot;
## [191] &quot;AGAP000814&quot; &quot;AGAP000815&quot; &quot;AGAP000821&quot; &quot;AGAP000823&quot; &quot;AGAP000825&quot;
## [196] &quot;AGAP000826&quot; &quot;AGAP000829&quot; &quot;AGAP000832&quot; &quot;AGAP000833&quot; &quot;AGAP000834&quot;
## [201] &quot;AGAP000835&quot; &quot;AGAP028655&quot; &quot;AGAP013506&quot; &quot;AGAP000840&quot; &quot;AGAP000841&quot;
## [206] &quot;AGAP000844&quot; &quot;AGAP000847&quot; &quot;AGAP000854&quot; &quot;AGAP029672&quot; &quot;AGAP000861&quot;
## [211] &quot;AGAP000862&quot; &quot;AGAP000863&quot; &quot;AGAP000894&quot; &quot;AGAP000901&quot; &quot;AGAP000929&quot;
## [216] &quot;AGAP012964&quot; &quot;AGAP000958&quot; &quot;AGAP000965&quot; &quot;AGAP000974&quot; &quot;AGAP000998&quot;
## [221] &quot;AGAP001015&quot; &quot;AGAP000413&quot; &quot;AGAP000519&quot;</code></pre>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
