*Anopheles* Genome Environment Associations 
==================================================================================

This repository contains a comprehensive compilation of code and expalantory visualization associated with the paper: Genome-environment Association Methods Comparison Indicates Omnigenic Adaptation to Ecological Niche in Malaria Vector Mosquitoes

Visualizations of processes described in the paper
------------

### Confirm convergence of BayeScEnv MCMC chains using the R package coda
*   For all 25 datasets (5 variables x 5 chromosome arms):
    > <https://devonderaad.github.io/Anopheles/run.bayescenv/bayescenv.confirm.convergence.html>

### Optimize mtry for each independent r2VIM run
*   Example for humidity ~ chromosome 2L:
    > <https://devonderaad.github.io/Anopheles/r2vim/optimize.mtry/hum.2L.r2vim.optimize.html>

### Perform 'repeated correlations' simple, intuitive GEA testing approach
*   Perform each correlation, store relevant values, explore significance cutoffs:
    > <https://devonderaad.github.io/Anopheles/xxx>

### Vetting process using allele frequency patterns from independent genetic sampling efforts
*   Entire process can be viewed here, including visualizations of the resulting classifications showing intuitive usefulness of this process:
    > <https://devonderaad.github.io/Anopheles/vetting/simple.vetting.all.GEAs.html>

### Testing the effect of spatial and genomic autocorrelation of dependent variable on GEA results 
*   The process of calculating correlation matrices, mantel tests, and evaluating the effect on number/proportion of GEA outliers can be viewed here:
    > <https://devonderaad.github.io/Anopheles/test.correlations/test.corrs.html>

### Identify sets of genes associated with each outlier SNP dataset, and test for significant overrepresentation of the a priori identified desiccation-tolerance candidate genes from Ayala et al. (2019)

*   This process can be viewed and reproduced from this markdown:
    > <https://devonderaad.github.io/Anopheles/xxx.html>

Access candidate gene datasets
------------
*   All 20 candidate gene and outlier SNP datasets generated here can be directly accessed by following:
    > <https://github.com/DevonDeRaad/Anopheles/tree/master/candidate.gene.datasets>


