---
title: "Precipitation associated genes of interest"
output:
  html_document:
    df_print: paged
---


```{r}
# package
library(ape)
library(clusterProfiler)

#bring in phase1 allele frequencies
phase1.all.freqs<-read.csv(file = "~/Desktop/anoph.3.march.2020/phase1.all.freqs.csv")
#make column for id
phase1.all.freqs$id<-paste(phase1.all.freqs$chrom,phase1.all.freqs$POS)
#bring in dataframe with locality and environmental data for each individual phase1
phase1.alldata<-read.csv(file = "~/Desktop/anoph.3.march.2020/phase1.allvariables.csv")
#subset to only the variables we need (lat,long,hum,temp,precip)
meta<-phase1.alldata[,c("latitude","longitude","pannual","tmean","hannual")]
#subset only the unique lat/longs
metapops<-unique(meta)
#sort by latitude to match the order of the allele frequency files
sort.pops.phase1 <-metapops[order(metapops$latitude),]

## read in lfmm outlier table
lfmm.outlier.table <- read.csv("~/Downloads/ml.outputs/vetted.lfmm.hum.env.csv")[,2:3]
#add separate chrom and pos columns
#add separate chrom and pos columns
#lfmm.outlier.table$chrom<-data.frame(do.call('rbind', strsplit(as.character(lfmm.outlier.table$id),' ',fixed=TRUE)))[,1]
#lfmm.outlier.table$pos<-data.frame(do.call('rbind', strsplit(as.character(lfmm.outlier.table$id),' ',fixed=TRUE)))[,2]

## read in bayescan outlier table
bayescan.outlier.table <- read.csv("~/Downloads/ml.outputs/vetted.bayescenv.hum.env.csv")
#add separate chrom and pos columns
#bayescan.outlier.table$chrom<-data.frame(do.call('rbind', strsplit(as.character(bayescan.outlier.table$id),' ',fixed=TRUE)))[,1]
#bayescan.outlier.table$pos<-data.frame(do.call('rbind', strsplit(as.character(bayescan.outlier.table$id),' ',fixed=TRUE)))[,2]

## read in r2vim outlier table
r2vim.outlier.table <- read.csv("~/Downloads/ml.outputs/vetted.r2vim.hum.env.csv")[,2:3]
#add separate chrom and pos columns
colnames(r2vim.outlier.table)<-c("chrom","pos")

## read in genomic regions and attributes file
genomic_location <- read.gff("~/Downloads/Anopheles_gambiae.AgamP4.47.chr.gff3.gz")
```


```{r}
#keep only regions identified as "gene"
gen_regions <- genomic_location[genomic_location$type == "gene", ] # subset to only gene regions

#pull all genes queried
all.gene.id<-data.frame(do.call('rbind', strsplit(as.character(gen_regions$attributes),';',fixed=TRUE)))[,1]
all.genes<-data.frame(do.call('rbind', strsplit(as.character(all.gene.id),':',fixed=TRUE)))[,2]
#write.table(all.genes, "~/Downloads/all.genes.anopheles.gambiae.txt", row.names = F, col.names = F, quote = F)
```


```{r}
#define chroms
chroms <- c("2R", "2L", "3R", "3L","X") # chromosomes of interest

#initialize gene.list for lfmm
lfmm.gene.list<-data.frame()
#fill gene.list by matching location of genes of interest in gene feature regions by chromosomes
for (i in 1:length(chroms)){
  genloc <- gen_regions[gen_regions$seqid == chroms[i], ] # genome info per chromosome
  pos <- as.numeric(as.character(lfmm.outlier.table[lfmm.outlier.table$chrom == chroms[i],2])) # outlierFST position per chromosome
  for (w in 1:length(pos)) {
    if (nrow(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,]) !=0){
    lfmm.gene.list<-rbind(lfmm.gene.list, cbind(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,], pos[w]))
    }
  }
  cat("chromosome", i, "of", length(chroms), "finished\n\n")
}

#pull out gene ID as its own column
lfmm.gene.list$gene.id<-data.frame(do.call('rbind', strsplit(as.character(lfmm.gene.list$attributes),';',fixed=TRUE)))[,1]
lfmm.gene.list$gene<-data.frame(do.call('rbind', strsplit(as.character(lfmm.gene.list$gene.id),':',fixed=TRUE)))[,2]

#make SNP ID column
lfmm.gene.list$SNPid<-paste(lfmm.gene.list$seqid,lfmm.gene.list$`pos[w]`)

#subset
lfmm.gene.list<-lfmm.gene.list[,c("SNPid","gene")]

#number of candidate snps
nrow(lfmm.outlier.table)

#calculate percentage of lfmm candidate SNPs that were in a gene
nrow(lfmm.gene.list)/nrow(lfmm.outlier.table)

#calc % of the genome covered by the genes we are searching
sum(gen_regions$end-gen_regions$start)/278268413 

#calc number of unique genes
length(unique(lfmm.gene.list$gene)) #830

table(lfmm.gene.list$gene)[table(lfmm.gene.list$gene) >10]
hist(table(lfmm.gene.list$gene))
#write.table(unique(lfmm.gene.list$gene), "~/Downloads/lfmm.prec.gene.list.txt", row.names = F, col.names = F, quote = F)
write.table(lfmm.gene.list, "/Users/devder/Desktop/anoph.phase2/overrep/lfmm.hum.gene.snp.match.txt", row.names = F, col.names = T, quote = F)
```

```{r}
#initialize gene.list for bayescan
bayescan.gene.list<-data.frame()
#fill gene.list
for (i in 1:length(chroms)){
  genloc <- gen_regions[gen_regions$seqid == chroms[i], ] # genome info per chromosome
  pos <- as.numeric(as.character(bayescan.outlier.table[bayescan.outlier.table$chrom == chroms[i],2])) # outlierFST position per chromosome
  for (w in 1:length(pos)) {
    if (nrow(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,]) !=0){
      bayescan.gene.list<-rbind(bayescan.gene.list, cbind(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,], pos[w]))
    }
  }
  cat("chromosome", i, "of", length(chroms), "finished\n\n")
}

#pull out gene ID as its own column
bayescan.gene.list$gene.id<-data.frame(do.call('rbind', strsplit(as.character(bayescan.gene.list$attributes),';',fixed=TRUE)))[,1]
bayescan.gene.list$gene<-data.frame(do.call('rbind', strsplit(as.character(bayescan.gene.list$gene.id),':',fixed=TRUE)))[,2]

#make SNP ID column
bayescan.gene.list$SNPid<-paste(bayescan.gene.list$seqid,bayescan.gene.list$`pos[w]`)

#subset
bayescan.gene.list<-bayescan.gene.list[,c("SNPid","gene")]

#number of candidate snps
nrow(bayescan.outlier.table)

#calculate percentage of bayescan candidate SNPs that were in a gene
nrow(bayescan.gene.list)/nrow(bayescan.outlier.table)

#calc % of the genome covered by the genes we are searching
sum(gen_regions$end-gen_regions$start)/278268413 

table(bayescan.gene.list$gene)[table(bayescan.gene.list$gene) >10]
hist(table(bayescan.gene.list$gene))

#calc number of unique genes
length(unique(bayescan.gene.list$gene)) #272

#write.table(unique(bayescan.gene.list$gene), "~/Downloads/bayescan.prec.gene.list.txt", row.names = F, col.names = F, quote = F)
write.table(bayescan.gene.list, "/Users/devder/Desktop/anoph.phase2/overrep/bayescan.hum.gene.snp.match.txt", row.names = F, col.names = T, quote = F)
#look into how many genes overlap between the methods
length(intersect(unique(lfmm.gene.list$gene),unique(bayescan.gene.list$gene)))
```

```{r}
#initialize gene.list for r2vim
r2vim.gene.list<-data.frame()
#fill gene.list
for (i in 1:length(chroms)){
  genloc <- gen_regions[gen_regions$seqid == chroms[i], ] # genome info per chromosome
  pos <- as.numeric(as.character(r2vim.outlier.table[r2vim.outlier.table$chrom == chroms[i],2])) # outlier position per chromosome
  for (w in 1:length(pos)) {
    if (nrow(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,]) !=0){
      r2vim.gene.list<-rbind(r2vim.gene.list, cbind(genloc[pos[w] >= genloc$start & pos[w] <= genloc$end,], pos[w]))
    }
  }
  cat("chromosome", i, "of", length(chroms), "finished\n\n")
}

#pull out gene ID as its own column
r2vim.gene.list$gene.id<-data.frame(do.call('rbind', strsplit(as.character(r2vim.gene.list$attributes),';',fixed=TRUE)))[,1]
r2vim.gene.list$gene<-data.frame(do.call('rbind', strsplit(as.character(r2vim.gene.list$gene.id),':',fixed=TRUE)))[,2]

#make SNP ID column
r2vim.gene.list$SNPid<-paste(r2vim.gene.list$seqid,r2vim.gene.list$`pos[w]`)

#subset
r2vim.gene.list<-r2vim.gene.list[,c("SNPid","gene")]

#number of candidate snps
nrow(r2vim.outlier.table)

#calculate percentage of bayescan candidate SNPs that were in a gene
nrow(r2vim.gene.list)/nrow(r2vim.outlier.table)

#calc % of the genome covered by the genes we are searching
sum(gen_regions$end-gen_regions$start)/278268413 

table(r2vim.gene.list$gene)[table(r2vim.gene.list$gene) >10]
hist(table(r2vim.gene.list$gene))

#calc number of unique genes
length(unique(r2vim.gene.list$gene)) #272

#write.table(unique(bayescan.gene.list$gene), "~/Downloads/bayescan.prec.gene.list.txt", row.names = F, col.names = F, quote = F)
write.table(r2vim.gene.list, "/Users/devder/Desktop/anoph.phase2/overrep/r2vim.hum.gene.snp.match.txt", row.names = F, col.names = T, quote = F)
#look into how many genes overlap between the methods
length(intersect(unique(lfmm.gene.list$gene),unique(bayescan.gene.list$gene)))
```

LFMM overrep testing
```{r}
#overrepresentation testing
#read in each GO term to gene map (col1=geneID,col2=GOterm)
mart.export<-read.table("~/Downloads/mart_export (3).txt", sep = "\t", header = T)

#calculate overrepresentation for lfmm prec
lfmm.prec.enriched<-enricher(gene = lfmm.gene.list$gene,
                 universe = all.genes,
                   pAdjustMethod = "fdr",
                 TERM2GENE= mart.export[,c(2,1)]
)
result<-lfmm.prec.enriched@result

lfmm.prec.enriched.sig<-result[result$p.adjust <= .05,]
lfmm.prec.enriched.sig
#there are 9 significantly enriched GO terms
write.table(result, file="/Users/devder/Desktop/anoph.phase2/overrep/lfmm.hum.go.overrep.txt",
            row.names = F, col.names = F, quote = F)

```

Bayescenv overrep testing
```{r}
#calculate overrepresentation for bayescan prec
bayescan.prec.enriched<-enricher(gene = bayescan.gene.list$gene,
                            universe = all.genes,
                            pAdjustMethod = "fdr",
                            TERM2GENE= mart.export[,c(2,1)]
                            )
result<-bayescan.prec.enriched@result

#retain significantly enriched GO terms
bayescan.prec.enriched.sig<-result[result$p.adjust <= .05,]

#check out significantly overrepresented GO terms
bayescan.prec.enriched.sig
write.table(result, file="/Users/devder/Desktop/anoph.phase2/overrep/bayescan.hum.go.overrep.txt",
            row.names = F, col.names = F, quote = F)

```

R2vim overrep testing
```{r}
#calculate overrepresentation for bayescan prec
r2vim.prec.enriched<-enricher(gene = r2vim.gene.list$gene,
                            universe = all.genes,
                            pAdjustMethod = "fdr",
                            TERM2GENE= mart.export[,c(2,1)]
                            )
result<-r2vim.prec.enriched@result

#retain significantly enriched GO terms
r2vim.prec.enriched.sig<-result[result$p.adjust <= .05,]

#check out significantly overrepresented GO terms
r2vim.prec.enriched.sig
write.table(result, file="/Users/devder/Desktop/anoph.phase2/overrep/r2vim.hum.go.overrep.txt",
            row.names = F, col.names = F, quote = F)

```

