---
title: "Simple.vetting.all.GEAs"
author: "Devon DeRaad"
date: "4/8/2021"
output: html_document
---

```{r}
#plot results from phase1 and verify with phase2 allele freq patterns
library(scattermore)
library(gridExtra)
library(VennDiagram)
library(qvalue)
library(outliers)
library(e1071)

#read in all sig values
all.sigs<-read.csv("~/Downloads/all.sigs.csv")
head(all.sigs)

#bring in phase1 allele frequencies
phase1.all.freqs<-read.csv(file = "~/Desktop/anoph.phase2/phase1.all.freqs.csv")
#make column for id
phase1.all.freqs$id<-paste(phase1.all.freqs$chrom,phase1.all.freqs$POS)
head(phase1.all.freqs)

#bring in phase2 allele frequencies
phase2.all.freqs<-read.csv(file = "~/Desktop/anoph.phase2/phase2.all.freqs.csv")
#make id column
phase2.all.freqs$id<-paste(phase2.all.freqs$chrom,phase2.all.freqs$POS)

#bring in dataframe with locality and environmental data for each individual phase1
phase1.alldata<-read.csv(file="~/Downloads/phase1_all_vairables_last.csv")
#subset to only the variables we need (lat,long,hum,temp,precip)
meta<-phase1.alldata[,c("latitude","longitude","p_annual","t_mean","h_mean","autocorrelated","random")]
#subset only the unique lat/longs
metapops<-unique(meta)
#sort by latitude to match the order of the allele frequency files
sort.pops.phase1 <-metapops[order(metapops$latitude),]

#bring in dataframe with locality and environmental data for each individual phase2
phase2.alldata<-read.csv(file = "~/Downloads/phase2_all_vairables_last.csv")
#subset to only the variables we need (lat,long,hum,temp,precip)
meta<-phase2.alldata[,c("latitude","longitude","p_annual","t_mean","h_mean","autocorrelated","random")]
#subset only the unique lat/longs
metapops<-unique(meta)
#sort by latitude to match the order of the allele frequency files
sort.pops.phase2<-metapops[order(metapops$latitude),]
sort.pops.phase2<-sort.pops.phase2[c(3,8,10:28),]

```

#make dataframes where each row contains allele frequency data for the given SNP and the environmental values used to generate significant GEA test for that SNP (for phase 1 data)
```{r}
#pull all significant outlier SNP IDs for LFMM humidity
out<- all.sigs[all.sigs$lfmm.hum < .01,c(1,2)]
#add the allele frequency at all 14 sites for each outlier SNP to the dataframe
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
#initialize empty dataframe
z<-data.frame()
#make matching dataframe with the value of the environmental variable used for testing for each SNP
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$h_mean)}
#fix column names to be consistent
colnames(z)<-c(sort.pops.phase1$latitude)
#combine into a single data.frame
lfmm.hum.outliers<-cbind(x,z,test=rep("lfmm.hum", times=nrow(x)))
#each row contains the identifier for an outlier SNP, the allele frequency of that SNP across phase1,
#the environmental values used for testing that SNP, and the name of the test

#lfmm precipitation
out<- all.sigs[all.sigs$lfmm.prec < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$p_annual)}
colnames(z)<-c(sort.pops.phase1$latitude)
lfmm.prec.outliers<-cbind(x,z,test=rep("lfmm.prec", times=nrow(x)))

#lfmm temperature
out<- all.sigs[all.sigs$lfmm.temp < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$t_mean)}
colnames(z)<-c(sort.pops.phase1$latitude)
lfmm.temp.outliers<-cbind(x,z,test=rep("lfmm.temp", times=nrow(x)))

#lfmm spatially autocorrelated variable
out<- all.sigs[all.sigs$lfmm.corr < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$autocorrelated)}
colnames(z)<-c(sort.pops.phase1$latitude)
lfmm.corr.outliers<-cbind(x,z,test=rep("lfmm.corr", times=nrow(x)))

#lfmm random variable
out<- all.sigs[all.sigs$lfmm.rand < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$random)}
colnames(z)<-c(sort.pops.phase1$latitude)
lfmm.rand.outliers<-cbind(x,z,test=rep("lfmm.rand", times=nrow(x)))

#bayescenv prec 
out<- all.sigs[all.sigs$baye.prec < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$p_annual)}
colnames(z)<-c(sort.pops.phase1$latitude)
baye.prec.outliers<-cbind(x,z,test=rep("baye.prec", times=nrow(x)))

#bayescenv hum
out<- all.sigs[all.sigs$baye.hum < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$h_mean)}
colnames(z)<-c(sort.pops.phase1$latitude)
baye.hum.outliers<-cbind(x,z,test=rep("baye.hum", times=nrow(x)))

#bayescenv temp
out<- all.sigs[all.sigs$baye.temp < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$t_mean)}
colnames(z)<-c(sort.pops.phase1$latitude)
baye.temp.outliers<-cbind(x,z,test=rep("baye.temp", times=nrow(x)))

#bayescenv corr
out<- all.sigs[all.sigs$baye.corr < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$autocorrelated)}
colnames(z)<-c(sort.pops.phase1$latitude)
baye.corr.outliers<-cbind(x,z,test=rep("baye.corr", times=nrow(x)))

#bayescenv rand 
out<- all.sigs[all.sigs$baye.rand < .01,c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$random)}
colnames(z)<-c(sort.pops.phase1$latitude)
baye.rand.outliers<-cbind(x,z,test=rep("baye.rand", times=nrow(x)))

#vim prec
out<- all.sigs[all.sigs$vim.prec > 1, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$p_annual)}
colnames(z)<-c(sort.pops.phase1$latitude)
vim.prec.outliers<-cbind(x,z,test=rep("vim.prec", times=nrow(x)))

#vim hum
out<- all.sigs[all.sigs$vim.hum > 1, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$h_mean)}
colnames(z)<-c(sort.pops.phase1$latitude)
vim.hum.outliers<-cbind(x,z,test=rep("vim.hum", times=nrow(x)))

#vim temp
out<- all.sigs[all.sigs$vim.temp > 1, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$t_mean)}
colnames(z)<-c(sort.pops.phase1$latitude)
vim.temp.outliers<-cbind(x,z,test=rep("vim.temp", times=nrow(x)))

#vim corr
out<- all.sigs[all.sigs$vim.corr > 1, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$autocorrelated)}
colnames(z)<-c(sort.pops.phase1$latitude)
vim.corr.outliers<-cbind(x,z,test=rep("vim.corr", times=nrow(x)))

#vim rand
out<- all.sigs[all.sigs$vim.rand > 1, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$random)}
colnames(z)<-c(sort.pops.phase1$latitude)
vim.rand.outliers<-cbind(x,z,test=rep("vim.rand", times=nrow(x)))

#cors prec
out<- all.sigs[all.sigs$cors.prec < .0001, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$p_annual)}
colnames(z)<-c(sort.pops.phase1$latitude)
cors.prec.outliers<-cbind(x,z,test=rep("cors.prec", times=nrow(x)))

#cors hum
out<- all.sigs[all.sigs$cors.hum < .0001, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$h_mean)}
colnames(z)<-c(sort.pops.phase1$latitude)
cors.hum.outliers<-cbind(x,z,test=rep("cors.hum", times=nrow(x)))

#cors temp
out<- all.sigs[all.sigs$cors.temp < .0001, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$t_mean)}
colnames(z)<-c(sort.pops.phase1$latitude)
cors.temp.outliers<-cbind(x,z,test=rep("cors.temp", times=nrow(x)))

#cors corr
out<- all.sigs[all.sigs$cors.corr < .0001, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$autocorrelated)}
colnames(z)<-c(sort.pops.phase1$latitude)
cors.corr.outliers<-cbind(x,z,test=rep("cors.corr", times=nrow(x)))

#cors rand
out<- all.sigs[all.sigs$cors.rand < .0001, c(1,2)]
x <- phase1.all.freqs[phase1.all.freqs$id %in% paste(out[,1],out[,2]),c(1:16)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase1$random)}
colnames(z)<-c(sort.pops.phase1$latitude)
cors.rand.outliers<-cbind(x,z,test=rep("cors.rand", times=nrow(x)))

#combine all the dataframes
full<-rbind(lfmm.hum.outliers,lfmm.prec.outliers,lfmm.temp.outliers,lfmm.rand.outliers,lfmm.corr.outliers,
            baye.hum.outliers,baye.prec.outliers,baye.temp.outliers,baye.rand.outliers,baye.corr.outliers,
            vim.hum.outliers,vim.prec.outliers,vim.temp.outliers,vim.rand.outliers,vim.corr.outliers,
            cors.hum.outliers,cors.prec.outliers,cors.temp.outliers,cors.rand.outliers,cors.corr.outliers)
```


```{r}
#build correlation dataframe for phase1 data
store<-c() #init vector
for (i in 1:nrow(full)){
  store[i]<-cor(as.numeric(as.vector(full[i,17:30])), as.numeric(as.vector(full[i,3:16]))) #calculate correlation coefficient for each SNP
}

#add correlation coefficient to df
full$cor<-store
```

#make dataframes where each row contains allele frequency data for the given SNP and the environmental values used to generate significant GEA test for that SNP (for phase 2 data)

```{r}
#####step2 repeat this process with phase2 data
####
###
##
#

#pull all significant outlier SNP IDs for LFMM humidity
out<- all.sigs[all.sigs$lfmm.hum < .01,c(1,2)]
#add the allele frequency at all 14 sites for each outlier SNP to the dataframe
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
#initialize empty dataframe
z<-data.frame()
#make matching dataframe with the value of the environmental variable used for testing for each SNP
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$h_mean)}
#fix column names to be consistent
colnames(z)<-c(sort.pops.phase2$latitude)
#combine into a single data.frame
lfmm.hum.outliers<-cbind(x,z,test=rep("lfmm.hum", times=nrow(x)))
#each row contains the identifier for an outlier SNP, the allele frequency of that SNP across phase1,
#the environmental values used for testing that SNP, and the name of the test

#lfmm precipitation
out<- all.sigs[all.sigs$lfmm.prec < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$p_annual)}
colnames(z)<-c(sort.pops.phase2$latitude)
lfmm.prec.outliers<-cbind(x,z,test=rep("lfmm.prec", times=nrow(x)))

#lfmm temperature
out<- all.sigs[all.sigs$lfmm.temp < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$t_mean)}
colnames(z)<-c(sort.pops.phase2$latitude)
lfmm.temp.outliers<-cbind(x,z,test=rep("lfmm.temp", times=nrow(x)))

#lfmm spatially autocorrelated variable
out<- all.sigs[all.sigs$lfmm.corr < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$autocorrelated)}
colnames(z)<-c(sort.pops.phase2$latitude)
lfmm.corr.outliers<-cbind(x,z,test=rep("lfmm.corr", times=nrow(x)))

#lfmm random variable
out<- all.sigs[all.sigs$lfmm.rand < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$random)}
colnames(z)<-c(sort.pops.phase2$latitude)
lfmm.rand.outliers<-cbind(x,z,test=rep("lfmm.rand", times=nrow(x)))

#bayescenv prec 
out<- all.sigs[all.sigs$baye.prec < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$p_annual)}
colnames(z)<-c(sort.pops.phase2$latitude)
baye.prec.outliers<-cbind(x,z,test=rep("baye.prec", times=nrow(x)))

#bayescenv hum
out<- all.sigs[all.sigs$baye.hum < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$h_mean)}
colnames(z)<-c(sort.pops.phase2$latitude)
baye.hum.outliers<-cbind(x,z,test=rep("baye.hum", times=nrow(x)))

#bayescenv temp
out<- all.sigs[all.sigs$baye.temp < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$t_mean)}
colnames(z)<-c(sort.pops.phase2$latitude)
baye.temp.outliers<-cbind(x,z,test=rep("baye.temp", times=nrow(x)))

#bayescenv corr
out<- all.sigs[all.sigs$baye.corr < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$autocorrelated)}
colnames(z)<-c(sort.pops.phase2$latitude)
baye.corr.outliers<-cbind(x,z,test=rep("baye.corr", times=nrow(x)))

#bayescenv rand 
out<- all.sigs[all.sigs$baye.rand < .01,c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$random)}
colnames(z)<-c(sort.pops.phase2$latitude)
baye.rand.outliers<-cbind(x,z,test=rep("baye.rand", times=nrow(x)))

#vim prec
out<- all.sigs[all.sigs$vim.prec > 1, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$p_annual)}
colnames(z)<-c(sort.pops.phase2$latitude)
vim.prec.outliers<-cbind(x,z,test=rep("vim.prec", times=nrow(x)))

#vim hum
out<- all.sigs[all.sigs$vim.hum > 1, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$h_mean)}
colnames(z)<-c(sort.pops.phase2$latitude)
vim.hum.outliers<-cbind(x,z,test=rep("vim.hum", times=nrow(x)))

#vim temp
out<- all.sigs[all.sigs$vim.temp > 1, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$t_mean)}
colnames(z)<-c(sort.pops.phase2$latitude)
vim.temp.outliers<-cbind(x,z,test=rep("vim.temp", times=nrow(x)))

#vim corr
out<- all.sigs[all.sigs$vim.corr > 1, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$autocorrelated)}
colnames(z)<-c(sort.pops.phase2$latitude)
vim.corr.outliers<-cbind(x,z,test=rep("vim.corr", times=nrow(x)))

#vim rand
out<- all.sigs[all.sigs$vim.rand > 1, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$random)}
colnames(z)<-c(sort.pops.phase2$latitude)
vim.rand.outliers<-cbind(x,z,test=rep("vim.rand", times=nrow(x)))

#cors prec
out<- all.sigs[all.sigs$cors.prec < .0001, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$p_annual)}
colnames(z)<-c(sort.pops.phase2$latitude)
cors.prec.outliers<-cbind(x,z,test=rep("cors.prec", times=nrow(x)))

#cors hum
out<- all.sigs[all.sigs$cors.hum < .0001, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$h_mean)}
colnames(z)<-c(sort.pops.phase2$latitude)
cors.hum.outliers<-cbind(x,z,test=rep("cors.hum", times=nrow(x)))

#cors temp
out<- all.sigs[all.sigs$cors.temp < .0001, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$t_mean)}
colnames(z)<-c(sort.pops.phase2$latitude)
cors.temp.outliers<-cbind(x,z,test=rep("cors.temp", times=nrow(x)))

#cors corr
out<- all.sigs[all.sigs$cors.corr < .0001, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$autocorrelated)}
colnames(z)<-c(sort.pops.phase2$latitude)
cors.corr.outliers<-cbind(x,z,test=rep("cors.corr", times=nrow(x)))

#cors rand
out<- all.sigs[all.sigs$cors.rand < .0001, c(1,2)]
x <- phase2.all.freqs[phase2.all.freqs$id %in% paste(out[,1],out[,2]),c(1:23)]
z<-data.frame()
for (i in 1:nrow(x)){z<-rbind(z,sort.pops.phase2$random)}
colnames(z)<-c(sort.pops.phase2$latitude)
cors.rand.outliers<-cbind(x,z,test=rep("cors.rand", times=nrow(x)))

#combine all the dataframes
full.2<-rbind(lfmm.hum.outliers,lfmm.prec.outliers,lfmm.temp.outliers,lfmm.rand.outliers,lfmm.corr.outliers,
              baye.hum.outliers,baye.prec.outliers,baye.temp.outliers,baye.rand.outliers,baye.corr.outliers,
              vim.hum.outliers,vim.prec.outliers,vim.temp.outliers,vim.rand.outliers,vim.corr.outliers,
              cors.hum.outliers,cors.prec.outliers,cors.temp.outliers,cors.rand.outliers,cors.corr.outliers)
```

#calculate correlation coefficient for phase 2 data
```{r}
#build correlation dataframe for phase2 data
store<-c() #init vector
for (i in 1:nrow(full.2)){
  store[i]<-cor(as.numeric(as.vector(full.2[i,24:44])), as.numeric(as.vector(full.2[i,3:23]))) #perform linear reg
}

#add as column in full.2
full.2$cor<-store
```

#Test each SNP for matching correlation coefficient and < .5 difference in magnitude
```{r}
#subset corframe 1 SNPs found in cor.frame 2
hom.full <- full[paste(full[,1],full[,2],full[,31]) %in% paste(full.2[,1],full.2[,2],full.2[,45]),]

#check to make sure all cells are homologous between these data frames
table(paste(hom.full$chrom,hom.full$pos,hom.full$test) == paste(full.2$chrom,full.2$pos,full.2$test))

#absolute value difference in correlation coefficient
hist(abs(hom.full$cor-full.2$cor))

# test how many SNPs have mismatched directional relationships between phase 1 and 2
table(hom.full$cor > 0 & full.2$cor < 0 | hom.full$cor < 0 & full.2$cor > 0)

#make dataframe containing info and vetting results for each SNP
vetted<-data.frame(chrom=hom.full$chrom,
                   pos=hom.full$POS,
                   test=hom.full$test,
                   pass.direction.test=hom.full$cor < 0 & full.2$cor < 0 | hom.full$cor > 0 & full.2$cor > 0,
                   pass.strength.test=abs(hom.full$cor-full.2$cor) < .5)

```

#estimate false positive rate overall and for each candidate SNP dataset
```{r}
#calculate overall false positive rate
1-sum(vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/nrow(vetted) #total tested SNPs

#because we expect our random variable (random between phases 1 and 2) to inflate the overall false positive rate,
#calculate false positive rate for only the real environmental variables
vet<-vetted[vetted$test != "lfmm.rand" & vetted$test != "baye.rand" & vetted$test != "vim.rand"
            & vetted$test != "lfmm.corr" & vetted$test != "baye.corr" & vetted$test != "vim.corr"
            & vetted$test != "cors.rand" & vetted$test != "cors.corr",]
1-sum(vet$pass.direction.test == TRUE & vet$pass.strength.test == TRUE)/nrow(vet) #total tested SNPs

#lfmm calculate number of successfully vetted SNPs and the false positive rate for each test
#lfmm hum
sum(vetted$test == "lfmm.hum") #total SNPs
sum(vetted$test == "lfmm.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "lfmm.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "lfmm.hum") #false positive rate

#lfmm prec
sum(vetted$test == "lfmm.prec") #total SNPs
sum(vetted$test == "lfmm.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "lfmm.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "lfmm.prec") #false positive rate

#lfmm temp
sum(vetted$test == "lfmm.temp") #total SNPs
sum(vetted$test == "lfmm.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "lfmm.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "lfmm.temp") #false positive rate

#lfmm rand
sum(vetted$test == "lfmm.rand") #total SNPs
sum(vetted$test == "lfmm.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "lfmm.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "lfmm.rand") #false positive rate

#lfmm corr
sum(vetted$test == "lfmm.corr") #total SNPs
sum(vetted$test == "lfmm.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "lfmm.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "lfmm.corr") #false positive rate

#baye calculate number of successfully vetted SNPs and the false positive rate for each test
#baye hum
sum(vetted$test == "baye.hum") #total SNPs
sum(vetted$test == "baye.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "baye.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "baye.hum") #false positive rate

#baye prec
sum(vetted$test == "baye.prec") #total SNPs
sum(vetted$test == "baye.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "baye.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "baye.prec") #false positive rate

#baye temp
sum(vetted$test == "baye.temp") #total SNPs
sum(vetted$test == "baye.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "baye.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "baye.temp") #false positive rate

#baye rand
sum(vetted$test == "baye.rand") #total SNPs
sum(vetted$test == "baye.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "baye.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "baye.rand") #false positive rate

#baye corr
sum(vetted$test == "baye.corr") #total SNPs
sum(vetted$test == "baye.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "baye.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "baye.corr") #false positive rate


#vim calculate number of successfully vetted SNPs and the false positive rate for each test
#vim hum
sum(vetted$test == "vim.hum") #total SNPs
sum(vetted$test == "vim.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "vim.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "vim.hum") #false positive rate

#vim prec
sum(vetted$test == "vim.prec") #total SNPs
sum(vetted$test == "vim.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "vim.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "vim.prec") #false positive rate

#vim temp
sum(vetted$test == "vim.temp") #total SNPs
sum(vetted$test == "vim.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "vim.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "vim.temp") #false positive rate

#vim rand
sum(vetted$test == "vim.rand") #total SNPs
sum(vetted$test == "vim.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "vim.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "vim.rand") #false positive rate

#vim corr
sum(vetted$test == "vim.corr") #total SNPs
sum(vetted$test == "vim.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "vim.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "vim.corr") #false positive rate

#cors calculate number of successfully vetted SNPs and the false positive rate for each test
#cors hum
sum(vetted$test == "cors.hum") #total SNPs
sum(vetted$test == "cors.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "cors.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "cors.hum") #false positive rate

#cors prec
sum(vetted$test == "cors.prec") #total SNPs
sum(vetted$test == "cors.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "cors.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "cors.prec") #false positive rate

#cors temp
sum(vetted$test == "cors.temp") #total SNPs
sum(vetted$test == "cors.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "cors.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "cors.temp") #false positive rate

#cors rand
sum(vetted$test == "cors.rand") #total SNPs
sum(vetted$test == "cors.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "cors.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "cors.rand") #false positive rate

#cors corr
sum(vetted$test == "cors.corr") #total SNPs
sum(vetted$test == "cors.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE) #nSNPs
1-sum(vetted$test == "cors.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE)/sum(vetted$test == "cors.corr") #false positive rate

#add column saying whether passed both
vetted$vet<-vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE
vetted$vet[vetted$vet == TRUE]<-"pass"
vetted$vet[vetted$vet == FALSE]<-"fail"

#write out vetted outlier dataset
write.csv(vetted, "~/Desktop/anoph.phase2/vetted.outliers.csv", row.names = F, quote=F)

#write out each vetted SNP dataset
write.csv(vetted[vetted$test == "lfmm.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/lfmm.hum.vetted.snps.csv" , quote=F, row.names = F) #nSNPs
write.csv(vetted[vetted$test == "lfmm.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/lfmm.prec.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "lfmm.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/lfmm.temp.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "lfmm.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/lfmm.rand.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "lfmm.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/lfmm.corr.vetted.snps.csv", quote=F, row.names = F) #nSNPs
write.csv(vetted[vetted$test == "baye.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/baye.hum.vetted.snps.csv", quote=F, row.names = F) #nSNPs
write.csv(vetted[vetted$test == "baye.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/baye.prec.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "baye.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/baye.temp.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "baye.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/baye.rand.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "baye.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/baye.corr.vetted.snps.csv", quote=F, row.names = F) #nSNPs
write.csv(vetted[vetted$test == "vim.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/vim.hum.vetted.snps.csv", quote=F, row.names = F)#nSNPs
write.csv(vetted[vetted$test == "vim.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/vim.prec.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "vim.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/vim.temp.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "vim.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/vim.rand.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "vim.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/vim.corr.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "cors.hum" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/cors.hum.vetted.snps.csv", quote=F, row.names = F) #nSNPs
write.csv(vetted[vetted$test == "cors.prec" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/cors.prec.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "cors.temp" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/cors.temp.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "cors.rand" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/cors.rand.vetted.snps.csv", quote=F, row.names = F) #nSNPs,]
write.csv(vetted[vetted$test == "cors.corr" & vetted$pass.direction.test == TRUE & vetted$pass.strength.test == TRUE,], "~/Downloads/vet.output/cors.corr.vetted.snps.csv", quote=F, row.names = F) #nSNPs

```

#plot 100 randomly selected points to assure visually that this vetting scheme is doing a satisfactory job in identifying SNPs where the putatively significant relationship identified by the GEA is not supported by the semi-independent phase 2 sampling (whether the given SNP passed vetting printed as the x axis label)
```{r}
#plot 100 random points to visually confirm that the classification worked
set.seed(62399)
plot.samples<-vetted[sample(nrow(vetted),100),]
par(mfrow=c(2,2))
for (i in paste(plot.samples$chrom,plot.samples$pos,plot.samples$test)){
  plot(as.numeric(as.vector(full[paste(full$chrom,full$POS,full$test) == i,c(17:30)])),
       as.numeric(as.vector(full[paste(full$chrom,full$POS,full$test) == i,c(3:16)])), main = i, ylim =c(0,1), xlab=plot.samples$vet[paste(plot.samples$chrom,plot.samples$pos,plot.samples$test) == i],ylab="allele frequency")
  abline(lm(as.numeric(as.vector(full[paste(full$chrom,full$POS,full$test) == i,c(3:16)]))~
              as.numeric(as.vector(full[paste(full$chrom,full$POS,full$test) == i,c(17:30)]))))
  points(as.numeric(as.vector(full.2[paste(full.2$chrom,full.2$POS,full.2$test) == i,c(24:44)])),
         as.numeric(as.vector(full.2[paste(full.2$chrom,full.2$POS,full.2$test) == i,c(3:23)])), col="red")
  abline(lm(as.numeric(as.vector(full.2[paste(full.2$chrom,full.2$POS,full.2$test) == i,c(3:23)]))~
              as.numeric(as.vector(full.2[paste(full.2$chrom,full.2$POS,full.2$test) == i,c(24:44)]))), col="red")
}
```

